enhancements/installer/bootstrap-ignition-shim-asset.md

```diff
@@ -0,0 +1,123 @@
+---
+title: Openshift Installer Bootstrap Ignition Shim
+authors:
+  - "@iamemilio"
+reviewers:
+  - TBD
+  - "@abhinavdahiya"
+  - "@wking"
+  - "@mandre"
+  - "@tomassedovic"
+approvers:
+  - TBD
+  - "@abhinavdahiya"
+  - "@wking"
+creation-date: 2019-10-17
+last-updated: 2019-10-17
+status: provisional
+---
+
+# Openshift Installer Bootstrap Ignition Shim Asset
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+There are a few platforms that use an Ignition shim for their bootstrap machine. The shim
+is essentially a pointer to the main config that is stored on a remote server, which helps
+us avoid problems with instance metadata size limits on our platform. Previously, we were doing
+this in Terraform using the terraform Ignition Provider, however the repo has fallen extremely out
+of date, making us unable to add additional CA Bundles to our nodes. This means that clients that
+use self signed certificates are unable to use our IPI installer, and can not upgrade to openshift 4 on Openstack.
+We considered both updating the
+terraform provider and creating a new installer asset as possible solutions, and ultimately decided
+that the best solution for our team is to create the installer asset. We plan to create an ignition
+asset similar to the [node asset](https://github.com/openshift/installer/blob/master/pkg/asset/ignition/machine/node.go)
+for bootstrap ignition shims that other platforms can **opt in** to.
+
+## Motivation
```

Comment:
So I see 3 points in this motivation:

- terraform-provider-ignition is stale and they are not taking on improvements. Which is the reason the Openstack team which is the heaviest user of the provider in terraform templates in the installer does not think they have the capacity to maintain a fork.
- CA bundle use requires ign >2.3 schema and because of reason above updating the provider to support this is probalematic
- Also the team wants to add `bootstrap-shim.ign` file to output of ignition-configs for all platforms as it would help Openstack UPI primarily but also other platforms.

correct?

Lets break this up into concise sections, one big flowing is difficult to read

Reply:
Yeah that seems right. I'll just break this into bullets